src/AspNetCoreMulti/Example/GraphQL/Operations.cs

```diff
@@ -0,0 +1,69 @@
+using Example.Repositories;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+
+namespace Example.GraphQL
+{
+    public interface IOperation
+    {
+        IEnumerable<IFieldType> RegisterFields();
+    }
+
+    public interface IDogOperation : IOperation { }
+
+    public interface ICatOperation : IOperation { }
+
+    public class DogOperation : ObjectGraphType<object>, IDogOperation
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public DogOperation(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public IEnumerable<IFieldType> RegisterFields()
+        {
+            var fields = new List<IFieldType>
+            {
+                Field<StringGraphType>("say", resolve: context => "woof woof woof"),
+                GetBreedListField(),
+                GetImageDetailsField()
+            };
+
+            return fields;
+        }
+
+        private IFieldType GetBreedListField()
+        {
+            return Field<NonNullGraphType<ListGraphType<NonNullGraphType<DogType>>>>("dogBreeds", resolve: context =>
+            {
+                using var scope = _serviceProvider.CreateScope();
+                var dogRepository = scope.ServiceProvider.GetRequiredService<DogRepository>();
+                var dogs = dogRepository.GetDogs();
+                return dogs;
+            });
+        }
+
+        private IFieldType GetImageDetailsField()
+        {
+            return FieldAsync<NonNullGraphType<ImageDetailsType>>("dogImageDetails", resolve: async context =>
+            {
+                using var scope = _serviceProvider.CreateScope();
+                var imageDetailsRepository = scope.ServiceProvider.GetRequiredService<DogImageDetailsRepository>();
+                var imageDetails = await imageDetailsRepository.GetDogImageDetails();
+                return imageDetails;
+            });
+        }
+    }
+
+    public class CatSayOperation : ObjectGraphType<object>, ICatOperation
+    {
+        public IEnumerable<IFieldType> RegisterFields()
+        {
+            return new List<IFieldType> { Field<StringGraphType>("say", resolve: context => "meow meow meow") };
```

Comment:
It looks rather awkward. You have added field via `Field<StringGraphType>` API here and then pass field up to register it into query type. I don't get it.

Reply:
I have changed this now. I am going for a functional approach and this sort of broke it I agree. I have now changed the way Im setting things up so it wont be added here but rather when the main query calls the method. The implementor would have to create the fields and return so they will be added to the query. Pls let me know this looks.